bundles/fr.adaussy.permadeler.model.design.tests/src/fr/adaussy/permadeler/model/design/services/ValidationServiceTest.java

```diff
@@ -0,0 +1,185 @@
+package fr.adaussy.permadeler.model.design.services;
+
+import static org.junit.Assert.assertTrue;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import fr.adaussy.permadeler.model.Permadeler.Area;
+import fr.adaussy.permadeler.model.Permadeler.CompatibilityLink;
+import fr.adaussy.permadeler.model.Permadeler.CompatibilityMatrix;
+import fr.adaussy.permadeler.model.Permadeler.Family;
+import fr.adaussy.permadeler.model.Permadeler.Genus;
+import fr.adaussy.permadeler.model.Permadeler.KnowledgeBase;
+import fr.adaussy.permadeler.model.Permadeler.PermadelerFactory;
+import fr.adaussy.permadeler.model.Permadeler.Plantation;
+import fr.adaussy.permadeler.model.Permadeler.Planting;
+import fr.adaussy.permadeler.model.Permadeler.Root;
+import fr.adaussy.permadeler.model.Permadeler.Species;
+import fr.adaussy.permadeler.model.Permadeler.Zone;
+
+public class ValidationServiceTest {
+
+	private static final PermadelerFactory sFactory = PermadelerFactory.eINSTANCE;
+
+	private static final int nbPlantations = 3;
+
+	private Root root = sFactory.createRoot();
```

Comment:
On a test class that will test several service please don't use attributes of the test class. Instead you may create a InnerClass that will hold the code required for your test.

Reply:
done